pkg/controller/managedcluster/managed_cluster_controller.go

```diff
@@ -90,8 +90,15 @@ func createKlusterletAddonConfig(client client.Client, cluster *mcv1.ManagedClus
 	err := client.Get(ctx, types.NamespacedName{Namespace: name, Name: name}, &kac)
 	if errors.IsNotFound(err) {
 		log.Info(fmt.Sprintf("Create a new KlusterletAddonConfig resource %s", name))
-		kacNew := newKlusterletAddonConfig(clusterType(cluster), name, hostedAddOnEnabled(cluster))
-		if kacNew == nil {
+		var kacNew *kacv1.KlusterletAddonConfig
```

Comment:
@zhiweiyin318 @zhujian7  This part may be more clear to view in here https://github.com/stolostron/klusterlet-addon-controller/blob/1d5f54a22ab4b7a729eccc98e2e8009a0cfdaec0/pkg/controller/managedcluster/managed_cluster_controller.go#L94